components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.server/src/main/java/org/wso2/carbon/identity/configuration/mgt/server/cors/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -0,0 +1,283 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.configuration.mgt.server.cors.internal.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementClientException;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Attribute;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceAdd;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceTypeAdd;
+import org.wso2.carbon.identity.configuration.mgt.server.cors.CORSManagementService;
+import org.wso2.carbon.identity.configuration.mgt.server.cors.exception.CORSServiceException;
+import org.wso2.carbon.identity.configuration.mgt.server.cors.internal.CORSManagementServiceHolder;
+import org.wso2.carbon.identity.configuration.mgt.server.cors.internal.function.CORSUrlToAttribute;
+import org.wso2.carbon.identity.configuration.mgt.server.cors.internal.function.CORSUrlToResourceAdd;
+import org.wso2.carbon.identity.configuration.mgt.server.cors.internal.function.ResourceToCORSUrl;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.configuration.mgt.server.cors.internal.Constants.CORS_URL_RESOURCE_NAME;
+import static org.wso2.carbon.identity.configuration.mgt.server.cors.internal.Constants.CORS_URL_RESOURCE_TYPE;
+
+/**
+ * Implementation of the CORSService.
+ */
+public class CORSManagementServiceImpl implements CORSManagementService {
+
+    private static final Log log = LogFactory.getLog(CORSManagementService.class);
+
+    @Override
+    public List<String> getCORSUrls(String tenantDomain) throws CORSServiceException {
+
+        try {
+            startTenantFlow(tenantDomain);
+
+            Resource resource = getConfigurationManager().getResource(CORS_URL_RESOURCE_TYPE, CORS_URL_RESOURCE_NAME);
```

Comment:
Does configuration manager supports caching

Reply:
Not yet. It has plans to support caching in the future. But it is not a priority at the moment. 
(@tharindu-bandara) 

@malithie The CORS Service should definitely have to cache support somewhere. Should we implement it here?

Reply:
Not yet. It has plans to support caching in the future. But it is not a priority at the moment. 
(@tharindu-bandara) 

@malithie The CORS Service should definitely have to caching support somewhere. Should we implement it here?

Reply:
**Note:** Cache will be added at the CORS valve.